src/cdk/tree/padding.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directionality} from '@angular/cdk/bidi';
+import {Directive, Input, Optional} from '@angular/core';
+import {CdkTreeNode} from './node';
+import {CdkTree} from './tree';
+
+/**
+ * Indent for the children tree nodes.
+ * This directive will add left-padding to the node to show hierarchy.
+ */
+@Directive({
+  selector: '[cdkTreeNodePadding]',
+  host: {
+    '[style.padding-left]': 'paddingIndentLeft()',
+    '[style.padding-right]': 'paddingIndentRight()',
+  }
+})
+export class CdkTreeNodePadding<T> {
+  /** The level of depth of the tree node. The padding will be `level * indent` pixels. */
+  @Input('cdkTreeNodePadding') level: number;
+
+  /** The indent for each level. Default number 40px from material design menu sub-menu spec. */
+  @Input('cdkTreeNodePaddingIndent') indent: number = 40;
+
+  constructor(private _treeNode: CdkTreeNode<T>,
+              private _tree: CdkTree<T>,
+              @Optional() private _dir: Directionality) {}
+
+  /** The padding indent value for the tree node. Returns a string with px numbers if not null. */
+  _paddingIndent(): string|null {
+    const nodeLevel = (this._treeNode.data && this._tree.treeControl.getLevel)
+      ? this._tree.treeControl.getLevel(this._treeNode.data)
+      : null;
+    const level = this.level || nodeLevel;
+    return level ? `${level * this.indent}px` : null;
+  }
+
+  /** The left padding indent value for the tree node. */
+  paddingIndentLeft(): string|null {
```

Comment:
Rather than having two properties and bindings for left and right, you could set the `padding-left`/`padding-right` instead. E.g.:

```
setPadding() {
  const padding = this._paddingIndent();
  const paddingProp = this._dir && this._dir.value === 'rtl' ? 'padding-right' : 'padding-left';
  
  this._renderer.setStyle(this._element.nativeElement, paddingProp, padding);
}
```

Reply:
Updated `padding.ts`. Now set the properties and listen to `dir.change`.